tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_rollback_test.dart

```diff
@@ -42,4 +42,97 @@ void main() async {
     expect(fetchedData.elementAtOrNull(0)?.number, data.number);
     expect(fetchedData.elementAtOrNull(1)?.number, data3.number);
   });
+
+  tearDown(() async {
+    await UniqueData.db.deleteWhere(session, where: (t) => Constant.bool(true));
+  });
+
+  test(
+      'Given serializable transaction settings when modifying the same value concurrently then one transaction is rolled back.',
+      () async {
+    var initialValue = 'initial';
+    var data = await UniqueData.db
+        .insertRow(session, UniqueData(number: 1, email: initialValue));
+
+    var settings = TransactionSettings(
+      isolationLevel: IsolationLevel.serializable,
+    );
+
+    var update = (int id, Session session, String email) async {
+      try {
+        await session.db.transaction((t) async {
+          var localData = await UniqueData.db.findById(
+            session,
+            id,
+            transaction: t,
+          );
+          await Future.delayed(Duration(milliseconds: 100));
+          await UniqueData.db.updateRow(
+            session,
+            localData!.copyWith(email: email),
+            transaction: t,
+          );
+        }, settings: settings);
+      } catch (e) {
+        return false;
+      }
+
+      return true;
+    };
+
+    var result = await Future.wait([
+      update(data.id!, session, 'bob'),
+      update(data.id!, session, 'alex'),
+      update(data.id!, session, 'jim'),
+      update(data.id!, session, 'hank'),
+    ]);
+
+    var success = result.where((e) => e).length;
+    expect(success, 1);
+    var dataAfter = await UniqueData.db.findById(session, data.id!);
+    expect(dataAfter?.email, isNot(initialValue));
```

Comment:
Question: What is the expected behavior here? One of the update will be successful and the others fails and we cannot know which one it will be?

Reply:
Exactly! We just want to validate that a single update is successful and that this has been committed to the database.